### PR TITLE
Feature/6412 dylan v2 user

### PR DESCRIPTION
**Pull request by Sparowhawk**
## Description of Change
added v2 user endpoint and deleted all v1 user endpoint usage. Also updated unit tests where needed. Will revamp/update unit tests in the unit test ticket I am working on

## Screenshots/Video
N/A

## Testing
yarn test

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations
AC:

basic user information is set via v2/user
this is done after all other front end tickets from the epic
any tests that are necessary

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

**Comment by Sparowhawk** [comment]
Please wait until Theo's ticket is merged to PR this one

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/AppealDetailsScreen/AppealStatus/AppealCurrentStatus/AppealCurrentStatus.test.tsx
Comment (theodur) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/AppealDetailsScreen/AppealStatus/AppealCurrentStatus/AppealCurrentStatus.test.tsx:47]: The `preloadedState` param can be removed also, since we're not setting the state anymore

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/AppealDetailsScreen/AppealStatus/AppealStatus.test.tsx
Comment (theodur) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/AppealDetailsScreen/AppealStatus/AppealStatus.test.tsx:37]: Same as my comment above, `preloadedState` is no longer needed, since`InitialState` will already be set by default

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimsHistoryScreen/ClaimsHistoryScreen.test.tsx
Comment (theodur) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimsHistoryScreen/ClaimsHistoryScreen.test.tsx:146]: `preloadedState` can be removed here as well

File: VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx
Comment (theodur) [inline VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx:40]: Another instance of `preloadedState` that can be removed

File: VAMobile/src/screens/HomeScreen/ProfileScreen/PersonalInformationScreen/PersonalInformationScreen.test.tsx
Comment (theodur) [inline VAMobile/src/screens/HomeScreen/ProfileScreen/PersonalInformationScreen/PersonalInformationScreen.test.tsx:73]: With the contact info PR merged, you can now initialize the data in the query cache via the [render function](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/6521804f5f754beb015253f92c1817587015f82f/VAMobile/src/testUtils.tsx#L246). Alternatively, you could also mock the API client to return this, but since the data will be prefetched by the time the user gets to this screen, initializing it in the query cache probably provides the more accurate experience

File: VAMobile/src/screens/HomeScreen/ProfileScreen/PersonalInformationScreen/PersonalInformationScreen.tsx
Comment (theodur) [inline VAMobile/src/screens/HomeScreen/ProfileScreen/PersonalInformationScreen/PersonalInformationScreen.tsx:35]: Instead of having this be a function, you can set `fullName` in the `getPersonalInformation` query function, similar to how we are with the formatted phone numbers in the contact info [query function](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/6521804f5f754beb015253f92c1817587015f82f/VAMobile/src/api/contactInformation/getContactInformation.tsx#L16-L21). This would avoid needing to import a function for all the screens where the full name is needed.

File: VAMobile/src/store/api/demo/mocks/personalInformation.json
Comment (theodur) [inline VAMobile/src/store/api/demo/mocks/personalInformation.json:8]:
```suggestion
          "middleName": null,
```
With the new endpoint, if `middleName` isn't set, it would be `null` instead of an empty string

File: VAMobile/src/screens/OnboardingCarousel/OnboardingCarousel.test.tsx
Comment (theodur) [inline VAMobile/src/screens/OnboardingCarousel/OnboardingCarousel.test.tsx:34]: `preloadedState` can be removed here as well

**Review by theodur — CHANGES_REQUESTED** [review]

File: VAMobile/src/api/personalInformation/queryKeys.ts
Comment (Sparowhawk) [reply VAMobile/src/api/personalInformation/queryKeys.ts:2]: updated

File: VAMobile/src/api/types/PersonalInformationData.ts
Comment (Sparowhawk) [reply VAMobile/src/api/types/PersonalInformationData.ts]: added

File: VAMobile/src/api/types/PersonalInformationData.ts
Comment (Sparowhawk) [reply VAMobile/src/api/types/PersonalInformationData.ts:11]: added

File: VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx:40]: removed

File: VAMobile/src/api/types/PersonalInformationData.ts
Comment (Sparowhawk) [reply VAMobile/src/api/types/PersonalInformationData.ts:23]: changed

File: VAMobile/src/screens/HomeScreen/ProfileScreen/PersonalInformationScreen/PersonalInformationScreen.tsx
Comment (Sparowhawk) [reply VAMobile/src/screens/HomeScreen/ProfileScreen/PersonalInformationScreen/PersonalInformationScreen.tsx:35]: updated, also did the same for birthdate

File: VAMobile/src/screens/HomeScreen/ProfileScreen/PersonalInformationScreen/PersonalInformationScreen.test.tsx
Comment (Sparowhawk) [reply VAMobile/src/screens/HomeScreen/ProfileScreen/PersonalInformationScreen/PersonalInformationScreen.test.tsx:73]: updated 6112 ticket with these notes. There is a lot of updating I have to do for this test file to make sure everything is working as it should so I feel this is fine for now and I will update it to the render function when I make the updates on that branch

**Review by dumathane — DISMISSED** [review]

File: VAMobile/src/api/personalInformation/getPersonalInformation.tsx
Comment (theodur) [inline VAMobile/src/api/personalInformation/getPersonalInformation.tsx:24]:
```suggestion
      birthDate: birthDay && formatDateMMMMDDYYYY(birthDay),
```
Makes it a little easier to read, and this won't call `formatDateMMMMDDYYYY` if `birthDate` is null

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/AppealDetailsScreen/AppealStatus/AppealCurrentStatus/AppealCurrentStatus.test.tsx
Comment (theodur) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/AppealDetailsScreen/AppealStatus/AppealCurrentStatus/AppealCurrentStatus.test.tsx:47]: Looks like this one was missed

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/AppealDetailsScreen/AppealStatus/AppealStatus.test.tsx
Comment (theodur) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/AppealDetailsScreen/AppealStatus/AppealStatus.test.tsx:37]: This one was also left behind

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimsHistoryScreen/ClaimsHistoryScreen.test.tsx
Comment (theodur) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimsHistoryScreen/ClaimsHistoryScreen.test.tsx:146]: Same here

File: VAMobile/src/screens/HomeScreen/VeteranStatusScreen/VeteranStatusScreen.tsx
Comment (theodur) [inline VAMobile/src/screens/HomeScreen/VeteranStatusScreen/VeteranStatusScreen.tsx:33]: Since this is only used in one place, we probably don't need to store it in a variable, and can just use `personalInfo?.fullName` directly

**Review by theodur — CHANGES_REQUESTED** [review]
Changes look good for the most part. I noticed a few things that need to be updated with the personal information types. Also, there are still a few strangling `preloadedState` params that can be removed, and a few minor nit picks. I mainly requested changes for the data types

File: VAMobile/src/api/personalInformation/getPersonalInformation.tsx
Comment (Sparowhawk) [reply VAMobile/src/api/personalInformation/getPersonalInformation.tsx:24]: updated

File: VAMobile/src/screens/HomeScreen/VeteranStatusScreen/VeteranStatusScreen.tsx
Comment (Sparowhawk) [reply VAMobile/src/screens/HomeScreen/VeteranStatusScreen/VeteranStatusScreen.tsx:33]: updated

File: VAMobile/src/api/types/PersonalInformationData.ts
Comment (Sparowhawk) [reply VAMobile/src/api/types/PersonalInformationData.ts:12]: updated

File: VAMobile/src/api/personalInformation/getPersonalInformation.tsx
Comment (Sparowhawk) [reply VAMobile/src/api/personalInformation/getPersonalInformation.tsx:23]: updated

File: VAMobile/src/api/types/PersonalInformationData.ts
Comment (theodur) [inline VAMobile/src/api/types/PersonalInformationData.ts:24]:
```suggestion
export type PersonalInformationData = PersonalInformationPayload['data']['attributes'] & {
  fullName: string
}
```

Another way this can be written

**Review by theodur — APPROVED** [review]
Changes look good!

**Review by dumathane — APPROVED** [review]

**Pull request merged**
